### PR TITLE
Add process documentation for contributing to APIv4

### PR DESCRIPTION
**Pull request by jwilander**
These checklists and processes should make contributing to APIv4 much easier and can later be converted into general API process docs once APIv4 is out the door.

It should already be listed in the sidebar as it's already covered by [this guide](https://github.com/mattermost/docs/blob/master/source/guides/developer.rst).

Feedback welcome, especially on placement and organization

### CODE REVIEW

File: source/developer/api4.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
+# APIv4 Contribution Processes
+
+Mattermost API version 4 is an active and in-progress project. We're looking for contributors to code server endpoints, build drivers and write documentation. If you're interested [please join our Mattermost instance](https://pre-release.mattermost.com/signup_user_complete/?id=f1924a8db44ff3bb41c96424cdc20676) and join the [APIv4 channel](https://pre-release.mattermost.com/core/channels/apiv4).
+
+This page will describe the processes for contributing to APIv4. If you have questions please ask in the [APIv4 channel](https://pre-release.mattermost.com/core/channels/apiv4). The project leads are Joram Wilander (@joram) on development and Jason Blais (@jason) on project management.
```
Comment (it33) [inline source/developer/api4.md]: Did we mean product manager or project manager?

Comment (jwilander) [inline source/developer/api4.md]: Sorry, meant product manager

**Comment by jasonblais** [comment]
Overall this looks great!

Converted to RST, made some text, grammar and formatting tweaks along the way. Ready for second PM review.

I think the location of the TOC is okay - although we should clean up our developer's guide before the next hackfest event.

**Comment by lfbrock** [comment]
+1, didn't add the ready to merge label yet because it seemed like some of the apiv4 files hadn't been merged into platform yet

**Comment by jasonblais** [comment]
Thanks @lfbrock! Added `Merge On Hold` label until https://github.com/mattermost/platform/pull/5191 is merged and I'll merge this PR after that.

**Pull request merged**